docs/create-custom-image-using-existing-docker-image.md

```diff
@@ -40,29 +40,53 @@ mvn clean install
 * Add the following lines to the Dockerfile:
 
 ```bash
-### Apply AMPs to the latest Community repository image.
-FROM alfresco/alfresco-content-repository-community:latest
+# Apply AMPs to the latest Community repository image.
+ARG BASE_IMAGE=alfresco/alfresco-content-repository-community
+ARG VERSION=latest
+
+FROM ${BASE_IMAGE}:${VERSION}
+
+# Need these viariables to give permissions to tomcat (refer Dockerfile below)
+ARG GROUPNAME=Alfresco
+ARG USERNAME=alfresco
+ARG TOMCAT_DIR=/usr/local/tomcat
+
+# Need to switch to root to add the new files
+USER root
 
 ### Copy the AMPs from build context to the appropriate location for your application server
-COPY amps /usr/local/tomcat/amps
+COPY amps ${TOMCAT_DIR}/amps
 
 ### Install AMPs on alfresco
-RUN java -jar /usr/local/tomcat/alfresco-mmt/alfresco-mmt*.jar install \
-              /usr/local/tomcat/amps/alfresco-bulk-import-*.amp /usr/local/tomcat/webapps/alfresco -nobackup -force
+RUN java -jar ${TOMCAT_DIR}/alfresco-mmt/alfresco-mmt*.jar install \
+              ${TOMCAT_DIR}/amps ${TOMCAT_DIR}/webapps/alfresco -directory -nobackup -force
+
+# Need to have all Tomcat files owned by root with group GROUPNAME as mentionned (refer Dockerfile below)
+RUN chgrp -R ${GROUPNAME} ${TOMCAT_DIR}/webapps && \
+    find ${TOMCAT_DIR}/webapps -type d -exec chmod 0750 {} \; && \
+    find ${TOMCAT_DIR}/webapps -type f -exec chmod 0640 {} \; && \
+    chmod -R g+r ${TOMCAT_DIR}/webapps && \
+    chgrp -R ${GROUPNAME} ${TOMCAT_DIR}
+
+# Switching back to alfresco user after having added amps files to run the container as non-root
+USER ${USERNAME}
+
 ```
 
+** Note: ** [Alfresco/acs-community-packaging/docker-alfresco/Dockerfile](https://github.com/Alfresco/acs-community-packaging/blob/master/docker-alfresco/Dockerfile).
```

Comment:
Please clarify what this link is for:

** Note ** Parent Dockerfile is located here: [Alfresco/acs-community-packaging/docker-alfresco/Dockerfile](https://github.com/Alfresco/acs-community-packaging/blob/master/docker-alfresco/Dockerfile)

Reply:
Can we put a link in the bash spot code?